### PR TITLE
Feat/injective with diff test in workflow

### PR DESCRIPTION
**Pull request by vic-en**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Aside the workflow which is run based on diff, the rest of the code in the pr is same as the original Injective  pr.

**Tests performed by the developer**:

**Tips for QA testing**:

### CODE REVIEW

**Review by fengtality — COMMENTED** [review]
While this issue won't block this review, I don't think the approach of adding new folders for every chain will scale. 

Going forward, any chain route that calls `poll` `balances` and other basic chains functions should use the existing routes in `/network` instead of creating their own routes. I will create a bounty for CoinAlpha to refactor the existing routes.

Screen Shot 2023-03-04 at 10 39 55 AM

File: docs/swagger/clob-routes.yml
Comment (fengtality) [inline docs/swagger/clob-routes.yml:202]: the following routes don't exist:
- GET orders/open
- GET orders/filled
- POST settleFunds

File: src/clob/clob.validators.ts
Comment (fengtality) [inline src/clob/clob.validators.ts:73]: please fix this typo so that variable is named `NETWORK_VALIDATIONS`

File: src/connectors/injective/injective.ts
Comment (fengtality) [inline src/connectors/injective/injective.ts:43]: Why is `LRUCache` needed to instantiate an Injective instance, whereas other connectors like Uniswap dont?

**Review by fengtality — DISMISSED** [review]
Please remove extraneous packages from `package.json` and fix the typos indicated, and then you can merge this in, and answer my questions about `LRUCache`.

I know there's a another Injective PR incoming so I am approving this one conditional on the fixes above. 

While this issue won't block this review, I don't think the approach of adding new folders for every chain will scale. 

This is not a blocker but something that I will address shortly:
---
Going forward, any chain route that calls `poll` `balances` and other basic chains functions should use the existing routes in `/network` instead of creating their own routes. I will create a bounty for CoinAlpha to refactor the existing routes.

Screen Shot 2023-03-04 at 10 39 55 AM

File: docs/swagger/clob-routes.yml
Comment (vic-en) [reply docs/swagger/clob-routes.yml:202]: Right.
They were created with the hope that they'll be used by Injective or future CLOB conector.
But I think it makes sense to remove them now. Connectors that require them in the future can add them.

File: src/connectors/injective/injective.ts
Comment (vic-en) [reply src/connectors/injective/injective.ts:43]: The cache in the case helps to control the number of instances that'll be held in memory.
It's not really a big deal considering the most likely use case by most users.
It can be removed if requested.

Comment (petioptrv) [reply src/connectors/injective/injective.ts:43]: The LRUCache is first added here, but really, all classes that manage instances using the class attribute dictionary approach are exposed to memory leaks and can incorporate it. As Victor pointed out, this is likely not an issue in practice, but this is a proposal to mitigate the edge cases. The con here, of course, is the additional dependency.

**Review by rapcmia — APPROVED** [review]
LGTM

- Successfully setup source client and gateway
- injective_protocol available on `gateway-list`
- Ran the following curl tests 
  - localhost connection ✅
  - `requests/add_injective_key.json`
  - `requests/injective_balances.json`
  - `requests/injective_poll.json`
  - `requests/injective_delete_order.json`
  - `requests/injective_transfer_to_bank.json`
  - `requests/injective_transfer_to_sub.json`

File: package.json
Comment (vic-en) [reply package.json:47]: They were required to remove the messages about unmet dependencies coming from Injective and exiting packages you raised in the initial pr - https://github.com/hummingbot/gateway/pull/19#issuecomment-1438765877

File: docs/swagger/clob-routes.yml
Comment (vic-en) [reply docs/swagger/clob-routes.yml:2]: removed

File: src/clob/clob.validators.ts
Comment (vic-en) [reply src/clob/clob.validators.ts:73]: resolved

File: package.json
Comment (fengtality) [reply package.json:47]: ok thanks!

**Comment by fengtality** [comment]
The Foundation will wait for further changes in this PR to handle upcoming Injective chain account changes before merging this in.

**Comment by vic-en** [comment]
@fengtality I think I address all the comments so far. Can you 'mark as resolved' those you think have been addressed?

File: src/connectors/injective/injective.ts
Comment (fengtality) [reply src/connectors/injective/injective.ts:43]: Thanks guys! I'll add `LRUCache` to Uniswap connector then

**Review by fengtality — APPROVED** [review]
LGTM - closing in favor of https://github.com/hummingbot/gateway/pull/51

**Pull request closed without merging**
